hydra/core/utils.py

```diff
@@ -79,13 +80,29 @@ def filter_overrides(overrides: Sequence[str]) -> Sequence[str]:
     return [x for x in overrides if not x.startswith("hydra.")]
 
 
+def get_callbacks(cfg: DictConfig) -> List[Callback]:
+    from hydra.utils import instantiate
+
+    callbacks = []
+    if OmegaConf.select(cfg, "hydra.callbacks"):
+        cfg = cfg.hydra.callbacks
+        callbacks = []
+        for c in cfg.keys():
+            callbacks.append(instantiate(OmegaConf.select(cfg, c)))
+
+    return callbacks
```

Comment:
select should be used when you want to do dynamic inspection.
there is nothing dynamic here:
```suggestion
      callbacks = []
      for callback in cfg.hydra.callbacks.values():
          callbacks.append(instantiate(callback))
    return callbacks
```

Reply:
The callbacks should also only be instantiated once.